src/main/java/com/avaya/letscode/CartItem.java

```diff
@@ -34,8 +34,16 @@ public int getCartItemValue() {
         }
         
         // if item has offer, return the item total based on offer
-        return (this.offer).getDiscountedValue(this.product, this.quantity);
+        return (this.offer).getDiscountedValue(this);
```

Comment:
We can refactor this in a tertiary operator...
return (this.offer != null) ? this.offer.getDiscountedValue(this) : getValueBeforeOffer(); // getValueBeforeOffer could be a private method as this is only needed inside this class.